addons/snailmail/controller/snailmail_webhook.py

```diff
@@ -0,0 +1,83 @@
+import hashlib
+import hmac
+import json
+import logging
+
+from odoo import http
+from odoo.http import request
+from odoo.tools import consteq
+
+_logger = logging.getLogger(__name__)
+
+LETTER_STATUS_MAPPING = {
+    'undeliverable': 'Letter was unable to be delivered',
+    'delivered': 'Letter is delivered successfully'
+}
+
+
+class SnailmailWebhookController(http.Controller):
+    @http.route('/webhook/snailmail/<string:event_type>', type='jsonrpc', auth='public', save_session=False)
+    def snailmail_webhook(self, event_type):
+        """ Receive and process letter sent by IAP pingen webhook. """
+
+        event = request.get_json_data()
```

Comment:
Why not using real parameters ? Is that a thing now to encapsulate everything and make controllers unreadable ?